Section header

JakeDawkins opened the pull request:
# Section Header
resolves #1524 

Added a section header with a heading on the left, and a component (button) on the right

@samclaridge what is the padding around this supposed to be? It looks like soft-half-quarter on the ends (not what I have in  the screenshot below). And the primary-dark here looks darker than the screenshot. It looks more like dark-tertiary

_note:_ the button is currently broken. btn--dark-primary shows the default green.

<img width="536" alt="screen shot 2016-12-04 at 5 25 19 pm" src="https://cloud.githubusercontent.com/assets/9259509/20869722/b0bcaad6-ba46-11e6-9aea-835ed158448b.png">


Yours: 
<img width="536" alt="" src="https://cloud.githubusercontent.com/assets/4967600/20810715/d736caa4-b7d8-11e6-8fb2-809d3bcc39dc.png">

jbaxleyiii commented:
**[Storybook Hub](https://hub.getstorybook.io/)** is building storybooks for every commit in this PR.

**Latest Storybook: https://2a6a1963-b155-4e19-8856-6fb35e153509.sbook.io/**
<sub>( Visit [here](https://hub.getstorybook.io/app/github.com/NewSpring/Holtzman/branch/section-header/) for more info )</sub>

> _This comment is automatically added by Storybook Hub_

samclaridge commented:
@JakeDawkins that's my bad, dark-primary was from a previous iteration. You're right the correct text color is dark-tertiary. Where you apply the padding is going to depend on how you structure the markup. The section header text should have padding left of 20px, or soft. The button small should have margin right of 10px. I imagined that those would be floating in a containing div that has 20px padding on the ends.

https://cl.ly/3C3A3H073V1u

jbaxleyiii commented on imports/components/sectionHeader/index.js:
```diff
@@ -0,0 +1,22 @@
+};
+
+const SectionHeader = ({ title, link }: ISectionHeader) =>
+  <div className="background--light-secondary one-whole">
```
@JakeDawkins let's leave the coloring to the parent component, so you can remove the background color here

JakeDawkins commented:
Updated. I removed the bg color, and fixed the padding 

<img width="693" alt="screen shot 2016-12-05 at 8 56 16 am" src="https://cloud.githubusercontent.com/assets/9259509/20887529/01f6b70c-bac9-11e6-8405-bf94550f2566.png">
<img width="517" alt="screen shot 2016-12-05 at 8 56 41 am" src="https://cloud.githubusercontent.com/assets/9259509/20887530/01f6c490-bac9-11e6-907c-ff7f0c1c3433.png">
<img width="625" alt="screen shot 2016-12-05 at 8 56 54 am" src="https://cloud.githubusercontent.com/assets/9259509/20887531/01f71256-bac9-11e6-84a5-6e8cf8fa2617.png">
<img width="705" alt="screen shot 2016-12-05 at 8 57 14 am" src="https://cloud.githubusercontent.com/assets/9259509/20887532/01f7a414-bac9-11e6-908d-11818f9c94f3.png">
